### ISSUE TITLE
typo of the configuration readme

### ISSUE THREAD

**Issue by emailtowalter** [SteeltoeOSS/Configuration#4]
Hi there... 2 minor issues to mention for future readers: 
-- The following code in the readme should be using "vcap:services" 
var instaneId = config["vcap:application:p-mysql:0:credentials:uri"] 

-- To make it work , I have to change the string and use "user-provided". i.e. "vcap:services:user-provided:0:credentials:uri" 

Lastly, I want to use poco object for the config in vcap service, and found CloudFoundryServicesOptions class is a collection and hard to parse. I ended up to write the following code. Welcome your suggestion/feedback.  

public void ConfigureServices(IServiceCollection services){....
             var configSection = GetVCAPService("my_config");
            services.Configure<MyConfigPoco>(configSection);
...}

 private IConfiguration GetVCAPService(string vcapServiceName)
        {
            var sections = Configuration.GetSection("vcap:services:user-provided");

            if (sections == null) return Configuration;

            foreach (var item in sections.GetChildren())
            {
                string name= (string)item.GetValue(typeof(string), "name");
                if (vcapServiceName == name)
                    return item.GetSection("credentials");
             }
            return Configuration;
        }

Thanks, 

Walter

**Comment by dgodd** [comment]
Hi Walter,

We have fixed your simple to address concerns:
* Fixing `vcap:application:` -> `vcap:services:` (where appropriate)
* Fixing the `instanceId` typo
* Replacing `p-mysql` with `[service-name]` to make its nature clear

Regarding the UX for reading values from the Configuration, we are not sure we have understood your concerns or what the best action to take is. For instance were you trying to do something similar to the following Linq ?
```csharp
config.GetSection("vcap:services:user-provided").GetChildren()
                .Where(x => x["name"] == "jim")
                .Select(x => x["credentials:uri"])
                .First()
```

Dave and Keaty

**Comment by emailtowalter** [comment]
I guess I am still old-fashion.I can use linq and inject into a poco class.
